src/modules/media/hooks/get-media-objects.ts

```diff
@@ -2,21 +2,39 @@ import { useQuery } from 'react-query';
 import { UseQueryResult } from 'react-query/types/react/types';
 
 import { MediaService } from '@media/services';
+import { ReadingRoomMediaType } from '@reading-room/types';
 import { QUERY_KEYS } from '@shared/const/query-keys';
 import { ApiResponseWrapper, ElasticsearchAggregations } from '@shared/types/api';
 
-import { MediaInfo } from '../types';
+import { MediaInfo, MediaTypes } from '../types';
 
 export function useGetMediaObjects(
 	filters:
 		| {
 				query: string | undefined;
+				format: ReadingRoomMediaType;
 		  }
 		| undefined,
 	page: number,
 	size: number
 ): UseQueryResult<ApiResponseWrapper<MediaInfo> & ElasticsearchAggregations> {
-	return useQuery([QUERY_KEYS.getMediaObjects, { filters, page, size }], () =>
-		MediaService.getAll(filters, page, size)
-	);
+	return useQuery([QUERY_KEYS.getMediaObjects, { filters, page, size }], () => {
+		const { format, ...rest } = filters || {};
+		const mediaFormat: MediaTypes | undefined =
+			format !== ReadingRoomMediaType.All ? format : undefined;
+
+		// TODO: improve (?)
+		// Run two queries to fetch aggregates across formats
+		return Promise.all([
+			MediaService.getAll({ ...rest, format: mediaFormat }, page, size),
+			MediaService.getAll(rest, page, size),
+		]).then((responses) => {
+			const [results, noFormat] = responses;
+
+			return {
+				...results,
+				aggregations: noFormat.aggregations,
+			};
+		});
+	});
```

Comment:
Currently when you switch between tabs you get a flash of no results:
![image](https://user-images.githubusercontent.com/1710840/157428712-8bf276fb-8ce7-4c06-9cb5-67f000d2bbc0.png)

You can avoid this by telling react-query to keep the previous results until the new ones are loaded. We might need to add some kind of indicator that the results are loading though, since with this change it could be that the user doesn't get enough feedback that his click was accepted.

```suggestion
		},
		{
			keepPreviousData: true,
		}
	);
```